### PR TITLE
samples: tfm: PSA firmware update sample

### PR DESCRIPTION
**Pull request by microbuilder**
This commit adds a sample application demonstrating how to use
the PSA Firmware Update API from TF-M. It also enables the
`FIRMWARE_UPDATE` partition to be included at build time.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>

### CODE REVIEW

File: samples/tfm_integration/psa_firmware/boards/nrf9160dk_nrf9160ns.overlay [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
```
Comment (ioannisg) [inline samples/tfm_integration/psa_firmware/boards/nrf9160dk_nrf9160ns.overlay:1]: @microbuilder I do not think you need this any more

File: samples/tfm_integration/psa_firmware/src/main.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		       info.version.iv_build_num);
+	} else {
+		printk("\nUnable to query active image version!\n");
+	}
+}
+
+void main(void)
+{
+	/* Initialize the TFM NS interface */
+	tfm_ns_interface_init();
```
Comment (ioannisg) [inline samples/tfm_integration/psa_firmware/src/main.c:68]: probably not needed either

**Comment by ioannisg** [comment]
Added 2.7 milestone here.

File: samples/tfm_integration/psa_firmware/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+
+# Override the binary used by qemu
+set(QEMU_KERNEL_OPTION "-device;loader,file=${CMAKE_BINARY_DIR}/tfm_merged.hex")
```
Comment (tejlmand) [inline samples/tfm_integration/psa_firmware/CMakeLists.txt:6]: is this still needed when boards themselves can define this. like here:
https://github.com/zephyrproject-rtos/zephyr/blob/67295be71ec389b7a52139facf8691552d92d4d6/boards/arm/mps2_an521/board.cmake#L15-L20

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: samples/tfm_integration/psa_firmware/CMakeLists.txt (hunk 3 above)
Comment (oyvindronningstad) [reply samples/tfm_integration/psa_firmware/CMakeLists.txt:6]: Shouldn't be needed at all now.

File: samples/tfm_integration/psa_firmware/src/main.c [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+/** Declare a reference to the application logging interface. */
+LOG_MODULE_DECLARE(app, CONFIG_LOG_DEFAULT_LEVEL);
+
+void fwu_disp_ver_active(void)
+{
+	psa_status_t status;
+	psa_image_info_t info = { 0 };
+
+	/* Image type can be one of: */
```
Comment (oyvindronningstad) [inline samples/tfm_integration/psa_firmware/src/main.c:22]: Nit: These comments can be formatted as

```
/* ...
 * ...
 * ...
 */

**Comment by microbuilder** [comment]
Removing 2.7.0 milestone since this won't make it in before code LTS2 freeze due to some issues with BL2 and the test platform being used.

**Comment by joerchan** [comment]
@microbuilder Could we maybe enable the TEST_NS_FWU tests in tfm_regression?

**Comment by microbuilder** [comment]
Closing in favor of #42466

**Pull request closed without merging**